File: Pycharm/TPJU/Bibliotheque/adresse.py
class Adresse:
    """
    Class representing an address
    """

    def __init__(self, rue: str = "", ville: str = "", code_postal: str = ""):
        """
        Constructor for objects of class Adresse

        Args:
            rue: The street name
            ville: The city name
            code_postal: The postal code
        """
        self._rue: str = rue
        self._ville: str = ville
        self._code_postal: str = code_postal

    @property
    def rue(self) -> str:
        """Get the street name"""
        return self._rue

    @rue.setter
    def rue(self, value: str) -> None:
        """Set the street name"""
        self._rue = value

    @property
    def ville(self) -> str:
        """Get the city name"""
        return self._ville

    @ville.setter
    def ville(self, value: str) -> None:
        """Set the city name"""
        self._ville = value

    @property
    def code_postal(self) -> str:
        """Get the postal code"""
        return self._code_postal

    @code_postal.setter
    def code_postal(self, value: str) -> None:
        """Set the postal code"""
        self._code_postal = value

    def get_full_address(self) -> str:
        """
        Get the complete formatted address

        Returns:
            The full address as a formatted string
        """
        return f"{self._rue}, {self._ville} {self._code_postal}"

File: BlueJ/TPJU_TEST_UNITAIRES/Adresse.java

/**
 * Write a description of class Adresse here.
 *
 * @author (your name)
 * @version (a version number or a date)
 */
public class Adresse
{
    private String rue;
    private String ville;
    private String codePostal;


    /**
     * Constructor for objects of class Adresse
     */
    public Adresse()
    {
        // initialise instance variables
    }

        // Getters and Setters
    public String getRue() {
        return rue;
    }

    public void setRue(String rue) {
        this.rue = rue;
    }

    public String getVille() {
        return ville;
    }

    public void setVille(String ville) {
        this.ville = ville;
    }

    public String getCodePostal() {
        return codePostal;
    }

    public void setCodePostal(String codePostal) {
        this.codePostal = codePostal;
    }
    
        // Method that returns the full address
    public String getFullAddress() {
        return rue + ", " + ville + " " + codePostal;
    }


}
